tests/MMS/non-uniform-boundaries/runtest

```diff
@@ -0,0 +1,148 @@
+            pf(*f[:-1],label=f[-1])
+        else:
+            pf(*f)
+        #plt.plot(mget(y),mget(f),'-o')
+        #plt.plot(mget(y),fo,'x-')
+        plt.legend()
+    timer = fig.canvas.new_timer(interval = 4000)
+    timer.add_callback(close_event)
+    timer.start()
+    plt.show()
+
+def create(nx,low):
+    udx=2*np.pi/nx
+    if not low:
+        ux=np.linspace(-1.5*udx,2*np.pi+1.5*udx,num=(nx+4))
+    else:
+        ux=np.linspace(-1*udx,2*np.pi+2*udx,num=(nx+4))
+    nux = ux-.8*np.cos(ux)
+    return nux
+
+def integrate(dat,low):
+    ret=np.zeros_like(dat)
+    cur=-dat[0]
+    cur-=dat[1]
+    if low:
+        for i in range(len(dat)):
+            ret[i] = cur
+            cur+=dat[i]
+    else:
+        for i in range(len(dat)):
+            cur+=dat[i]/2
+            ret[i] = cur
+            cur+=dat[i]/2
+    return ret
+
+
+mesh_cache={}
+def test(bdry, func, loc):
+    print("Testing %s with %s at %s"%(bdry,func,loc),end=' ... ')
+    errs=[]
+    dy=[]
+
+    for ny in np.logspace(start,mmax,num=mmax-start+1,base=2):
+        global mesh_cache
+        try:
+            mesh=mesh_cache[ny]
+        except KeyError:
+            bc.setOption("mesh:ny","%d"%ny,force=True)
+            bc.setOption("mesh:dy","(1+0.8*sin(y))/%d*2*pi"%ny,force=True)
+            mesh=bc.Mesh(section="mesh")
+            mesh_cache[ny]=mesh
+        mdy = mesh.coordinates.dy.get()[3,:].flatten()
+        y=integrate(mdy,loc != "CENTRE")
+        fd=func(y)
+        f = bc.create3D("y",mesh,outloc=loc)
+        fds= fd.reshape((1,int(ny)+4,1))
+        for i in range(6):
+            f[i,:,0] = fds
+
+        fo=mget(f)
+        opt=bc.Options("")
+
+        myg=int(opt.get("mesh:myg",0))
+
+        for i in range(myg):
+            f[0, i  ,0]=-1
+            f[0,-1-i,0]=-1
+
+        f.applyBoundary(bdry)
+        f=mget(f)
+        if doPlot > 2:
+            plot([y,f,"with bc"],[y,fo,"org"],[y,"numpy"],[integrate(mdy,loc != "CENTRE"),"bc"])
+        er=np.max(np.abs(f-fo))
+        errs.append(er)
+        dy.append(1./ny)
+
+    errc=np.log(errs[-2]/errs[-1])
+    difc=np.log(dy[-2]/dy[-1])
+    conv=errc/difc
+    success= order - .25 < conv < order + .25
+    if doPlot > 1 or (not success and doPlot > 0):
+        plot([dy,errs,"log","error vs spacing"])
+    if success:
+        print("Pass")
+        return 0
+    else:
+        print("Failure: %s %s %s - expected %d, got %g"%(bdry,func,loc,order,conv))
+        return 1
+
+
+
+testdata=[
+    ["dirichlet_nu_o4", np.sin, 5],
+    ["dirichlet_nu_o3", np.sin, 3],
+    ["dirichlet_nu_o2", np.sin, 3],
+    ["neumann_nu_o2", np.cos,2],
+    ["neumann_nu_o3", np.cos,[5,2]],
+    ["neumann_nu_o4", np.cos,[4,2]],
+    ["free_nu_o2", np.cos,2],
+    ["free_nu_o3", np.cos,[4,2]],
+    ["free_nu_o4", np.cos,[4,2]],
+    ["free_nu_o2", np.sin,[3,1]],
+    ["free_nu_o3", np.sin,[3,1]],
+    ["free_nu_o4", np.sin,[5,1]]
+]
+
+code=0
+locs=["CENTRE", "YLOW" ]#"XLOW", "YLOW", "ZLOW":]
```

Comment:
added.

It might make sense to extend this a bit, to take variations in x also into account.

Also running a test in x direction would be useful ...